src/main/java/com/box/sdk/BoxResourceIterable.java

```diff
@@ -0,0 +1,158 @@
+package com.box.sdk;
+
+import java.net.URL;
+import java.util.Iterator;
+import java.util.NoSuchElementException;
+
+import com.eclipsesource.json.JsonArray;
+import com.eclipsesource.json.JsonObject;
+
+/**
+ * Common implementation for paging support.
+ *
+ * @author Stanislav Dvorscak
+ *
+ * @param <T>
+ *            type of iterated entity
+ */
+public abstract class BoxResourceIterable<T> implements Iterable<T> {
```

Comment:
The Web Hook endpoint Get Webhooks is based  on a limit and marker scheme like a lot of the V2 endpoints.   While we agree that your new Iterator implementation is an elegant solution, we don't think it should be included as part of this pull request which should be focused only on providing a Java Implementation for the WebHooks API.  We do not want to one form for webhooks and another form for the other paged resources.   At some point point in the future (perhaps near future) we may want to add your implementation to all the other endpoints so please keep the code around.  However for this pull request please use the  JSONIterator interface.